src/crypto.rs

```diff
@@ -95,15 +95,22 @@ pub struct Blake3Hasher {
 }
 
 impl Blake3Hasher {
-    pub fn new(key: Vec<u8>) -> Self {
-        let mut new_key = [0u8; 32];
-        if key.len() == 32 {
-            new_key.copy_from_slice(&key[..32]);
-        } else {
-            panic!("Key must be 32 bytes long");
-        }
+    pub fn new(secret: Option<Vec<u8>>) -> Self {
+        secret.as_ref().inspect(|s| {
+            if s.len() < MIN_SECRET_SIZE {
+                panic!("Secret must be at least {MIN_SECRET_SIZE} bytes long");
+            }
+        });
+
+        // blake3 key must be exactly 32 bytes long
+        let mut key = [0u8; 32];
```

Comment:
I don't know if this is possible in Rust, but could we have something like `let mut key = [0u8; MIN_SECRET_SIZE]` instead? Just to make sure we stay consistent with the key size. The same applies for the lines bellow (lines 108, 109, 111)